src/components/App/App.component.jsx

```diff
@@ -23,11 +22,9 @@ function App() {
 
   return (
     <BrowserRouter>
-      <AuthProvider>
-        <Theme>
-          <Layout>{GetSwitch()}</Layout>
-        </Theme>
-      </AuthProvider>
+      <Theme>
+        <Layout>{GetSwitch()}</Layout>
```

Comment:
I would take the function `GetSwitch` into another component like: `Routes` with the content of the switch:

Routes.component.jsx
```js
function Routes () {
  <Switch>
  // All the routes
  <Switch/>
}

export default Routes;
```

App.component.jsx
```js
function App () {
  return (
    <BrowserRouter>
      <Layout>
        <Routes/>
      </Layout>
    </BrowserRouter>
  )
}
```

I give the suggestion because it is not very common to see functions inside functions, the only application I can remember of having nested functions is to use closures and this is not the case.

Reply:
That is a cleaner approach, I implemented and it worked just great, thanks!